app/controllers/catkeys_controller.rb

```diff
@@ -3,15 +3,44 @@
 class CatkeysController < ApplicationController
   load_and_authorize_resource :cocina, parent: false, class: 'Repository', id_param: 'item_id'
 
-  rescue_from Dor::Services::Client::UnexpectedResponse do |exception|
-    md = /\((.*)\)/.match exception.message
-    detail = JSON.parse(md[1])['errors'].first['detail']
-    redirect_to solr_document_path(params[:item_id]),
-                flash: { error: "Unable to retrieve the cocina model: #{detail.truncate(200)}" }
+  ### classes that define a virtual catkey model object and data structure, used in form editing...persistence is in the cocina model
+  class ModelProxy
+    def initialize(id:, catkeys:)
+      @id = id # the object ID
+      @catkeys = catkeys # the array of catkey objects (defined in custom class below)
+    end
+
+    attr_reader :id, :catkeys
+
+    def to_param
+      @id
+    end
+
+    def persisted?
+      true
+    end
   end
 
+  class CatkeyRow
```

Comment:
Can you move this into CatkeyForm?

Reply:
yes, same comment as below for ModelProxy